frontend/exporter/src/types/copied.rs

```diff
@@ -841,18 +861,60 @@ pub struct ExpnData {
 }
 
 /// Reflects [`rustc_span::Span`]
-#[derive(Clone, Debug, Serialize, Deserialize, JsonSchema, Eq, Ord)]
+#[derive(::serde::Serialize, ::serde::Deserialize, Clone, Debug, JsonSchema, Eq, Ord)]
 pub struct Span {
     pub lo: Loc,
     pub hi: Loc,
     pub filename: FileName,
     /// Original rustc span; can be useful for reporting rustc
     /// diagnostics (this is used in Charon)
+    #[cfg(feature = "rustc")]
     #[serde(skip)]
     pub rust_span_data: Option<rustc_span::SpanData>,
+    #[cfg(not(feature = "rustc"))]
+    #[serde(skip)]
+    pub rust_span_data: Option<()>,
     // expn_backtrace: Vec<ExpnData>,
 }
 
+impl bincode::Encode for Span {
```

Comment:
Why can't we derive these?

Reply:
We need to skip a field and the proc macros from bincode can't do that (see https://github.com/bincode-org/bincode/issues/452)

Reply:
Added a comment in the code